common/djangoapps/util/testing.py

```diff
@@ -28,16 +28,17 @@ class UrlResetMixin(object):
     Doing this is expensive, so it should only be added to tests that modify settings
     that affect the contents of urls.py
     """
-
+    URLS_AUTO_RESET = True
```

Comment:
Consider:
1. Changing the class variable to `URLS_REST_EVERY_TEST`.
2. Adding a `setUpClass`/`tearDownClass` method on this Mixin that automatically resets URLs when `URLS_REST_EVERY_TEST` is `False`.
3. Then updating `setUp` to reset URLs only when `URLS_REST_EVERY_TEST` is `True`

Advantage: Test classes that use this mixin no longer need to have their own `setUpClass` methods (as currently needed in this PR).  In case they used to override settings within their `setUp` method, that would move as a class decorator instead.